solver/src/solver/http_solver.rs

```diff
@@ -202,9 +206,47 @@ impl HttpSolver {
                     },
                     mandatory: false,
                 };
-                (*index, uniswap)
+                (*index, pool_model)
             })
-            .collect()
+            .collect();
+        let balancer_cost = self.balancer_cost(gas_price);
+        let weighted_product_models: HashMap<_, AmmModel> = weighted_product_orders
+            .iter()
+            .map(|(index, amm)| {
+                let reserves = amm
+                    .reserves
+                    .iter()
+                    .map(|(token, state)| {
+                        (
+                            *token,
+                            PoolTokenData {
+                                balance: state.balance,
+                                // Bfp -> BigRational -> String -> BigDecimal
+                                weight: BigDecimal::from_str(
+                                    &BigRational::from(state.weight).to_string(),
+                                )
+                                .unwrap(),
+                            },
+                        )
+                    })
+                    .collect();
+                let pool_model = AmmModel {
+                    parameters: AmmParameters::WeightedProduct(WeightedProductPoolParameters {
+                        reserves,
+                    }),
+                    fee: BigDecimal::from(*amm.fee.numer()) / BigDecimal::from(*amm.fee.denom()),
+                    cost: CostModel {
+                        amount: balancer_cost,
+                        token: self.native_token,
+                    },
+                    mandatory: false,
+                };
+                (*index + constant_product_models.len(), pool_model)
```

Comment:
This indexing should be tested if not already!

Reply:
Can you maybe add a comment here to clarify (so it doesn't surprise someone reading the code later?).

Reply:
Done!